openam-authentication/wrenam-auth-duo/src/main/java/org/wrensecurity/wrenam/authentication/modules/duo/Duo.java

```diff
@@ -98,28 +125,52 @@ public int process(Callback[] callbacks, int state) throws AuthLoginException {
                     throw new AuthLoginException(SERVICE_KEY, "missingTransactionId", null);
                 }
                 debug.message("Checking Cisco Duo authentication status for transaction '" + transactionId + "'.");
+                if (System.currentTimeMillis() > expireTime) {
```

Comment:
Not sure I follow the logic of `expirationTime`. It is being set to `pollingTimeout` and if the polling from the client is slower we fail the authentication? That does not seem right.

Reply:
See my comment below.